vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -2013,3 +2051,129 @@ resource "vsphere_virtual_machine" "vm" {
 		os.Getenv("VSPHERE_USE_LINKED_CLONE"),
 	)
 }
+
+func testAccResourceVSphereVirtualMachineConfigSlashNetwork() string {
+	return fmt.Sprintf(`
+variable "datacenter" {
+  default = "%s"
+}
+
+variable "hosts" {
+  default = [
+    "%s",
+    "%s",
+    "%s",
+  ]
+}
+
+variable "switch_nic" {
+  default = "%s"
+}
+
+variable "cluster" {
+  default = "%s"
+}
+
+variable "resource_pool" {
+  default = "%s"
+}
+
+variable "network_label" {
+  default = "%s"
+}
+
+variable "ipv4_address" {
+  default = "%s"
+}
+
+variable "ipv4_prefix" {
+  default = "%s"
+}
+
+variable "ipv4_gateway" {
+  default = "%s"
+}
+
+variable "datastore" {
+  default = "%s"
+}
+
+variable "template" {
+  default = "%s"
+}
+
+variable "linked_clone" {
+  default = "%s"
+}
+
+data "vsphere_datacenter" "datacenter" {
+  name = "${var.datacenter}"
+}
+
+data "vsphere_host" "host" {
+  count         = "${length(var.hosts)}"
+  name          = "${var.hosts[count.index]}"
+  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  count          = "${length(data.vsphere_host.host.*.id)}"
+  name           = "vSwitchTerraformTest"
+  host_system_id = "${data.vsphere_host.host.*.id[count.index]}"
+
+  network_adapters = ["${var.switch_nic}"]
+
+  active_nics  = ["${var.switch_nic}"]
+  standby_nics = []
+}
+
+resource "vsphere_host_port_group" "pg" {
+  count               = "${length(data.vsphere_host.host.*.id)}"
+  name                = "${var.network_label}"
+  host_system_id      = "${data.vsphere_host.host.*.id[count.index]}"
+  virtual_switch_name = "${vsphere_host_virtual_switch.switch.*.name[count.index]}"
+}
+
+resource "vsphere_virtual_machine" "vm" {
+  name          = "terraform-test"
+  datacenter    = "${var.datacenter}"
+  cluster       = "${var.cluster}"
+  resource_pool = "${var.resource_pool}"
+
+  vcpu   = 2
+  memory = 1024
+
+  network_interface {
+    label              = "${vsphere_host_port_group.pg.0.name}"
+    ipv4_address       = "${var.ipv4_address}"
+    ipv4_prefix_length = "${var.ipv4_prefix}"
+    ipv4_gateway       = "${var.ipv4_gateway}"
+  }
+
+  disk {
+    datastore = "${var.datastore}"
+    template  = "${var.template}"
+    iops      = 500
+  }
+
+  depends_on = ["vsphere_host_port_group.pg"]
+
+  linked_clone = "${var.linked_clone != "" ? "true" : "false" }"
+}
+`,
+		os.Getenv("VSPHERE_DATACENTER"),
+		os.Getenv("VSPHERE_ESXI_HOST"),
+		os.Getenv("VSPHERE_ESXI_HOST2"),
+		os.Getenv("VSPHERE_ESXI_HOST3"),
+		os.Getenv("VSPHERE_HOST_NIC0"),
+		os.Getenv("VSPHERE_CLUSTER"),
+		os.Getenv("VSPHERE_RESOURCE_POOL"),
+		testAccResourceVSphereVirtualMachineSlashNetLabel,
+		os.Getenv("VSPHERE_IPV4_ADDRESS"),
+		os.Getenv("VSPHERE_IPV4_PREFIX"),
+		os.Getenv("VSPHERE_IPV4_GATEWAY"),
+		os.Getenv("VSPHERE_DATASTORE"),
+		os.Getenv("VSPHERE_TEMPLATE"),
+		os.Getenv("VSPHERE_USE_LINKED_CLONE"),
```

Comment:
_minor/not a blocker_ feels like this lot would make a nice test helper object

Reply:
Yeah, things are getting a little ridiculous here :wink:

Definitely could be something we could assemble into an object, although I'm a little resistant to the idea of breaking the variable configuration off, which is kind of what this would do, as right now all config fixtures are pretty easy to read and look like real Terraform config rather than chunks, which was a problem with this test code before (in a different way, though, one that made a lot less sense than what would be accomplished with a proper test fixture generator. :slightly_smiling_face: )